rules/checks.js

```diff
@@ -0,0 +1,187 @@
+const { parse } = require('@connectedcars/react-i18n/dist/parser')
+
+const defaultOptions = {
+  globalData: [],
+  replaceStringRegex: {
+    pattern: '{__KEY__}',
+  },
+  expressions: {
+    t: ['singular', 'data', 'context'],
+    tn: ['count', 'singular', 'plural', 'data', 'context'],
+    tx: ['singular', 'data', 'context'],
+    tnx: ['count', 'singular', 'plural', 'data', 'context'],
+  },
+}
+
+module.exports = {
+  create(context) {
+    const options = {
+      ...defaultOptions,
+      ...context.options[0],
+    }
+
+    const strictPattern = '[\\w-]+'
+
+    return {
+      CallExpression(node) {
+        const funcName = node.callee.name
+
+        const args = options.expressions[funcName]
+        if (!args) {
+          return
+        }
+
+        const singularIndex = args.indexOf('singular')
+        const singular = node.arguments[singularIndex]
+        if (!singular) {
+          return context.report({
+            node,
+            message: 'Singular function defined but no singular string found',
+          })
+        }
+
+        const isPluralFunc = args.includes('plural')
+        const pluralIndex = args.indexOf('plural')
+        const plural = node.arguments[pluralIndex]
+        if (isPluralFunc && !plural) {
+          return context.report({
+            node,
+            message: 'Plural function defined but no plural string found',
+          })
+        }
+
+        const dataIndex = args.indexOf('data')
+        const dataArg = node.arguments[dataIndex]
+        if (dataArg && dataArg.type !== 'ObjectExpression') {
+          return context.report({
+            node,
+            message: 'Data argument must be an object',
```

Comment:
just curious, but would this show before the tsc type warning you would get in a ts project?

Reply:
This also happens when a variable is passed instead of a direct object (the `type` is `Identifier` in this case). I'd like to be able to support that, but not really sure how to do it yet as the AST doesn't seem to easily return the referenced identifier node, only the name of it, as far as I can see.

Reply:
I've just changed it to `Data argument must be an inline object` so that it's more clear.